### PR TITLE
[JBPM-9529] Business-central REST API is allowing container creation …

### PR DESCRIPTION
**Pull request by abhijithumbe**
…with empty container id

**Thank you for submitting this pull request**

**JIRA**: _(https://issues.redhat.com/browse/JBPM-9529)_

### CODE REVIEW

**Comment by sutaakar** [comment]
Jenkins retest this please

**Comment by sutaakar** [comment]
Jenkins retest this please

**Review by sutaakar — APPROVED** [review]
Looks good to me.
One generic comment - in case this or other similar exceptions are returned then the HTTP response contains XML or JSON content-type header, however the error content is returned as a plain string. May it be worth to adjust the implementation to return maybe text/plain content type in such case? Anyway this is out of the scope for this PR.

**Comment by sutaakar** [comment]
@elguardian Can you please take a look?

**Comment by abhijithumbe** [comment]
@sutaakar Thank you for reviewing it. About the response content-type header I will open new jira and try to fix it. 
Do I need to add unit test for containerUpdate functionality check ?

**Comment by sutaakar** [comment]
> Do I need to add unit test for containerUpdate functionality check ?

yes, you can

**Comment by abhijithumbe** [comment]
@sutaakar unit test for ContainerUpdate functionality added. please take a look.

**Review by sutaakar — APPROVED** [review]
Looks good to me

**Comment by sutaakar** [comment]
Jenkins retest this please

**Comment by sutaakar** [comment]
@abhijithumbe Can you squash the commits into one?

**Comment by abhijithumbe** [comment]
> @abhijithumbe Can you squash the commits into one?

done

> @abhijithumbe Can you squash the commits into one?

done

**Comment by sutaakar** [comment]
Jenkins retest this please

File: kie-server-parent/kie-server-controller/kie-server-controller-impl/src/main/java/org/kie/server/controller/impl/service/SpecManagementServiceImpl.java [hunk 1]
```diff
@@ -58,6 +58,10 @@ public synchronized void saveContainerSpec(String serverTemplateId,
             throw new KieServerControllerIllegalArgumentException("No server template found for id " + serverTemplateId);
         }
 
+        if (containerSpec.getId().isEmpty()) {
```
Comment (adrielparedes) [inline kie-server-parent/kie-server-controller/kie-server-controller-impl/src/main/java/org/kie/server/controller/impl/service/SpecManagementServiceImpl.java:61]: Can containerSpec or getID() return a null value?

File: kie-server-parent/kie-server-controller/kie-server-controller-impl/src/main/java/org/kie/server/controller/impl/service/SpecManagementServiceImpl.java [hunk 2]
```diff
@@ -94,6 +98,12 @@ public synchronized void updateContainerSpec(String serverTemplateId, String con
             throw new KieServerControllerIllegalArgumentException("No server template found for id " + serverTemplateId);
         }
 
+
+        if (containerSpec.getId().isEmpty()) {
```
Comment (adrielparedes) [inline kie-server-parent/kie-server-controller/kie-server-controller-impl/src/main/java/org/kie/server/controller/impl/service/SpecManagementServiceImpl.java:102]: Can containerSpec or getID() return a null value?

**Review by adrielparedes — CHANGES_REQUESTED** [review]
@abhijithumbe good job! I have just a few questions. If no null value then is ok for me. @ederign should we include QE?

File: kie-server-parent/kie-server-controller/kie-server-controller-impl/src/main/java/org/kie/server/controller/impl/service/SpecManagementServiceImpl.java (hunk 1 above)
Comment (abhijithumbe) [reply kie-server-parent/kie-server-controller/kie-server-controller-impl/src/main/java/org/kie/server/controller/impl/service/SpecManagementServiceImpl.java]: yes, its possible. Changes done.

**Comment by abhijithumbe** [comment]
@adrielparedes Please review the changes.

File: kie-server-parent/kie-server-controller/kie-server-controller-impl/src/main/java/org/kie/server/controller/impl/service/SpecManagementServiceImpl.java [hunk 3]
```diff
@@ -58,6 +58,10 @@ public synchronized void saveContainerSpec(String serverTemplateId,
             throw new KieServerControllerIllegalArgumentException("No server template found for id " + serverTemplateId);
         }
 
+        if (containerSpec.getId().isEmpty() || containerSpec.getId()==null ) {
```
Comment (adrielparedes) [inline kie-server-parent/kie-server-controller/kie-server-controller-impl/src/main/java/org/kie/server/controller/impl/service/SpecManagementServiceImpl.java:61]: In this case if getId() is null it will throw a NPE, you should check if `containerSpec.getId()==null` before checking `getId().isEmpty()`. Same in code below. To check what I'm saying add a small test just for null values and you will see that behavior

**Review by adrielparedes — CHANGES_REQUESTED** [review]

File: kie-server-parent/kie-server-controller/kie-server-controller-impl/src/main/java/org/kie/server/controller/impl/service/SpecManagementServiceImpl.java (hunk 3 above)
Comment (abhijithumbe) [reply kie-server-parent/kie-server-controller/kie-server-controller-impl/src/main/java/org/kie/server/controller/impl/service/SpecManagementServiceImpl.java]: Yes, agreed null check should come first.

**Review by adrielparedes — APPROVED** [review]
@abhijithumbe good job. 
@ederign

**Comment by sutaakar** [comment]
Jenkins retest this please

**Comment by ederign** [comment]
@abhijithumbe thanks for your contribution!
